arangod/Transaction/Methods.cpp

```diff
@@ -1532,10 +1545,9 @@ OperationResult transaction::Methods::insertLocal(
       // Now replicate the good operations on all followers:
       std::string path =
         "/_db/" + arangodb::basics::StringUtils::urlEncode(databaseName()) +
-        "/_api/document/" +
-        arangodb::basics::StringUtils::urlEncode(collection->name()) +
-        "?isRestore=true&isSynchronousReplication=" +
-        ServerState::instance()->getId();
+        "/_api/document/" + arangodb::basics::StringUtils::urlEncode(collection->name()) +
+        "?isRestore=true&isSynchronousReplication=" + ServerState::instance()->getId() +
+        "&" + StaticStrings::OverWrite + "=" + (options.overwrite ? "true" : "false"); //CHEKME
```

Comment:
Would it make more sense to use the PUT API, which is meant for replacements, instead of adding an extra option via URL to the creation API? I haven't looked too in-depth at what's there. We could also handle a more natural UPSERT with property merging via the PATCH API, maybe?

Reply:
We have discussed this long with @neunhoef and he was sure that POST is the thing to do.